wit/deps/spin-postgres@3.0.0/postgres.wit

```diff
@@ -0,0 +1,94 @@
+package spin:postgres@3.0.0;
+
+interface postgres {
+  /// Errors related to interacting with a database.
+  variant error {
+      connection-failed(string),
+      bad-parameter(string),
+      query-failed(string),
+      value-conversion-failed(string),
+      other(string)
+  }
+
+  /// Data types for a database column
+  enum db-data-type {
+      boolean,
+      int8,
+      int16,
+      int32,
+      int64,
+      floating32,
+      floating64,
+      str,
+      binary,
+      date,
+      time,
+      datetime,
+      timestamp,
+      other,
+  }
+
+  /// Database values
+  variant db-value {
+      boolean(bool),
+      int8(s8),
+      int16(s16),
+      int32(s32),
+      int64(s64),
+      floating32(float32),
+      floating64(float64),
+      str(string),
+      binary(list<u8>),
+      date(tuple<s32, u8, u8>), // (year, month, day)
+      time(tuple<u8, u8, u8, u32>), // (hour, minute, second, nanosecond)
+      datetime(tuple<s32, u8, u8, u8, u8, u8, u32>), // (year, month, day, hour, minute, second, nanosecond)
+      timestamp(s64), // Unix timestamp (seconds since epoch)
+      db-null,
+      unsupported,
+  }
+
+  /// Values used in parameterized queries
+  variant parameter-value {
+      boolean(bool),
+      int8(s8),
+      int16(s16),
+      int32(s32),
+      int64(s64),
+      floating32(float32),
+      floating64(float64),
+      str(string),
+      binary(list<u8>),
+      date(tuple<s32, u8, u8>), // (year, month, day)
+      time(tuple<u8, u8, u8, u32>), // (hour, minute, second, nanosecond)
```

Comment:
Strings at the WIT level?  No, these specifications follow @tyler-harpool's original PR, and I think mimic Postgres' own concept of what the types contain.  I'd be wary of strings at the WIT level because they're so flexible and their semantics so ambiguous, although I know Postgres allows e.g. `time('11:22:33')` in SQL so maybe it would be fine.  In my mind I'd expect adapters so users would pass things like Rust `chrono::NaiveDate` or JS `Date` rather than raw WIT fields but maybe I am overthinking it?  I am not sure

Reply:
`time('11:22:33')` this was the precise example i was thinking about that shows how permissive PG is and how it can parse many formats. By adapter do you mean SDKs on top of the WIT? Regardless, I am happy to leave this as is. We can always add a type with a string parameter if there are user requests

Reply:
Yes, SDKs or helper functions or whatever.  Like Rust's native Postgres driver accepts `chrono::NaiveDate` for dates, so it would be good to offer something like that.